.github/workflows/python-package.yml

```diff
@@ -35,11 +35,23 @@ jobs:
         # extend-select will add rules in addition to defaults
         flake8 . --extend-select=E9,F63,F7,F82,W605 --count --exit-zero --max-complexity=10 --statistics
   build:
-    runs-on: ubuntu-latest
     needs: formatting
+    runs-on: ${{ matrix.os }}
+    name: Build ${{ matrix.os }} - ${{ matrix.arch }}
```

Comment:
I think you want to add lxml/not and python version in here too

Reply:
I don't see the change? I mean you're going to want to include it as part of the "name", since right now if you look at the workflows, it looks like the same named workflow is running multiple times.